task-sdk/docs/concepts.rst

```diff
@@ -0,0 +1,100 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+Concepts
```

Comment:
I don’t think these concepts belong in the task sdk documentation. 

These docs should focus solely on the DAG author and interfaces relevant to them. 

Broader architectural or system level concepts are better suited for the core Airflow docs.

Reply:
Not really, I do not think it matters for task sdk

Reply:
Also, the concept of metadata access

Reply:
Updated it @amoghrajesh Please re-review

Reply:
I have moved relevant concepts to upgrading guide now. Please re-review